TiKV config: Add quota-related configurations (#7978)
pingcap/docs: tikv-configuration-file.md

+ Enables the in-memory pessimistic lock feature. With this feature enabled, pessimistic transactions try to store their locks in memory, instead of writing the locks to disk or replicating the locks to other replicas. This improves the performance of pessimistic transactions. However, there is a still low probability that the pessimistic lock gets lost and causes the pessimistic transaction commits to fail.
+ Default value: `true`
+ Note that `in-memory` takes effect only when the value of `pipelined` is `true`.

## quota

Configuration items related to Quota Limiter.

> Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if TiKV foreground processes requests in too large volumes, the foreground occupies the CPU resources for processing background requests, which causes performance degradation. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the volume of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).

Reviewer: "If the amount of requests processed by TiKV in the foreground is so large that it affects the CPU resources for background processing, performance will degrade." 

maybe the above more appropriate? @en-jin19

> Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if TiKV foreground processes requests in too large volumes, the foreground occupies the CPU resources for processing background requests, which causes performance degradation. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the volume of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).

Reviewer: In other words, the number is not the point, the point is whether the background can continue without being affected

> Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if TiKV foreground processes requests in too large volumes, the foreground occupies the CPU resources for processing background requests, which causes performance degradation. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the volume of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).

Author: ```suggestion
Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if the amount of requests processed by TiKV in the foreground is too large, the CPU resources for background processing are occupied by the foreground, causing TiKV performance to degrade. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the volume of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).
```

> Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if the amount of requests processed by TiKV in the foreground is too large, the CPU resources for background processing are occupied by the foreground, causing TiKV performance to degrade. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the amount of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).

Reviewer 2: ```suggestion
Suppose that your machine on which TiKV is deployed has limited resources, for example, 4v CPU, 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources in the background are used to help process such requests, affecting the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).
```

> Suppose that your machine deployed TiKV has limited resources, for example, 4v CPU, 16 G memory. In this case, if the amount of requests processed by TiKV in the foreground is too large, the CPU resources for background processing are occupied by the foreground, causing TiKV performance to degrade. To avoid this issue, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When requests trigger the Quota Limiter feature, the requests are forced to wait for a while to free up CPU resources. The exact waiting time depends on the amount of the requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-durationnew-in-v600).

Author: We generally use the format of adding a blank between a number and a unit (16 G) and revise contents that do not meet this rule. However, there are some special cases, and one of them is the format of values: if it is a value of a configuration item, we use 16G or 16 GB according to different circumstances.

> Suppose that your machine on which TiKV is deployed has limited resources, for example, 4v CPU, 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources in the background are used to help process such requests, affecting the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).

Author: ```suggestion
Suppose that your machine on which TiKV is deployed has limited resources, for example, 4v CPU, 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources required by the background are used to help process such requests, affecting the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).
```

> **Warning:**
>

> > - Quota Limiter is an experimental feature introduced TiDB v6.0.0, and it is **NOT** recommended to use it in the production environment.

Reviewer 2: ```suggestion
> - Quota Limiter is an experimental feature introduced in TiDB v6.0.0, and it is **NOT** recommended to use it in the production environment.
```

> - This feature is only suitable for environments with limited resources, which ensures that TiKV can run stably in those environments for the long term. If you enable this feature in an environment with rich resources, performance degradation might be caused when the amount of requests reaches a peak.

> ### `foreground-cpu-time`(new in v6.0.0)

Reviewer 2: ```suggestion
### `foreground-cpu-time` (new in v6.0.0)
```

+ The soft limit on controlling the CPU resources used by TiKV foreground to process read and write requests.

> + Default value: `0` (no limit)

Reviewer 2: ```suggestion
+ Default value: `0` (which means no limit)
```

> + Unit: millicpu (`1500` means that foreground requests consume 1.5v CPU)

Reviewer 2: ```suggestion
+ Unit: millicpu (for example, `1500` means that foreground requests consume 1.5v CPU)
```

> ### `foreground-write-bandwidth`(new in v6.0.0)

Author: ```suggestion
### `foreground-write-bandwidth` (new in v6.0.0)
```

+ The soft limit for transactions to write data.

> + Default value: `0KB` (no limit)

Reviewer 2: ```suggestion
+ Default value: `0KB` (which means no limit)
```

> ### `foreground-read-bandwidth`(new in v6.0.0)

Author: ```suggestion
### `foreground-read-bandwidth` (new in v6.0.0)
```

> + The soft limit for transactions and Coprocessor to read data.

Reviewer 2: ```suggestion
+ The soft limit on the bandwidth with which transactions and the Coprocessor read data.
```

> + Default value: `0KB` (no limit)

Author: ```suggestion
+ Default value: `0KB` (which means no limit)
```

> ### `max-delay-duration`(new in v6.0.0)

Author: ```suggestion
### `max-delay-duration` (new in v6.0.0)
```

+ Enables the in-memory pessimistic lock feature. With this feature enabled, pessimistic transactions try to store their locks in memory, instead of writing the locks to disk or replicating the locks to other replicas. This improves the performance of pessimistic transactions. However, there is a still low probability that the pessimistic lock gets lost and causes the pessimistic transaction commits to fail.
+ Default value: `true`
+ Note that `in-memory` takes effect only when the value of `pipelined` is `true`.

## quota

Configuration items related to Quota Limiter.

> Suppose that your machine on which TiKV is deployed has limited resources, for example, 4v CPU, 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources required by the background are used to help process such requests, affecting the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).

Reviewer 2: ```suggestion
Suppose that your machine on which TiKV is deployed has limited resources, for example, with only 4v CPU and16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources used by the background are occupied to help process such requests, which affects the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources to be used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).
```

> Suppose that your machine on which TiKV is deployed has limited resources, for example, with only 4v CPU and 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources used by the background are occupied to help process such requests, which affects the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources to be used by the foreground. When a request triggers the Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).

Author: ```suggestion
Suppose that your machine on which TiKV is deployed has limited resources, for example, with only 4v CPU and 16 G memory. In this situation, if the foreground of TiKV processes too many read and write requests, the CPU resources used by the background are occupied to help process such requests, which affects the performance stability of TiKV. To avoid this situation, you can use the quota-related configuration items to limit the CPU resources to be used by the foreground. When a request triggers Quota Limiter, the request is forced to wait for a while for TiKV to free up CPU resources. The exact waiting time depends on the number of requests, and the maximum waiting time is no longer than the value of [`max-delay-duration`](#max-delay-duration-new-in-v600).
```

> **Warning:**
>
> - Quota Limiter is an experimental feature introduced in TiDB v6.0.0, and it is **NOT** recommended to use it in the production environment.
> - This feature is only suitable for environments with limited resources to ensure that TiKV can run stably in those environments. If you enable this feature in an environment with rich resources, performance degradation might occur when the amount of requests reaches a peak.

### `foreground-cpu-time` (new in v6.0.0)

> + The soft limit on controlling the CPU resources used by TiKV foreground to process read and write requests.

Reviewer 2: ```suggestion
+ The soft limit on the CPU resources used by TiKV foreground to process read and write requests.
```
